### PR TITLE
Move to fastfield codec

### PR DESCRIPTION
**Pull request by fulmicoton**

### CODE REVIEW

File: fastfield_codecs/src/serialize.rs [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+fn codec_estimation<C: FastFieldCodec, D: Column>(
+    fastfield_accessor: &D,
+    estimations: &mut Vec<(f32, FastFieldCodecType)>,
+) {
+    if let Some(ratio) = C::estimate(fastfield_accessor) {
+        estimations.push((ratio, C::CODEC_TYPE));
+    }
+}
+
+/// Serialize data into a new u64 fast field. The best compression codec will be chosen
```
Comment (PSeitz) [inline fastfield_codecs/src/serialize.rs]: the comment doesn't fit the function

File: fastfield_codecs/src/serialize.rs [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+    let gcd_params_opt = if codecs.contains(&FastFieldCodecType::Gcd) {
+        gcd_params(&column)
+    } else {
+        None
+    };
+
+    let gcd_params = if let Some(gcd_params) = gcd_params_opt {
+        gcd_params
+    } else {
+        return serialize_without_gcd(column, output, codecs);
```
Comment (PSeitz) [inline fastfield_codecs/src/serialize.rs:143]: we could alternatively remove gcd from the codecs and call self

File: fastfield_codecs/src/serialize.rs [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+        }
+    }
+    output.flush()?;
+    Ok(())
+}
+
+pub fn serialize_and_load<T: MonotonicallyMappableToU64 + Ord + Default>(
+    column: &[T],
+) -> Arc<dyn Column<T>> {
+    let mut buffer = Vec::new();
```
Comment (PSeitz) [inline fastfield_codecs/src/serialize.rs:218]: call `serialize_and_load_given_codecs` with ALL_CODEC_TYPES

File: fastfield_codecs/src/serialize.rs (hunk 2 above)
Comment (fulmicoton) [reply fastfield_codecs/src/serialize.rs:143]: I agree! I think we need to remove gcd from the codecs and treat it differently. (somewhat like you use to do).

Have a footer common to all codecs with num_vals, min_val, max_val, and gcd. If gcd == 1 then we return a reader without any gcd mapper.

File: fastfield_codecs/src/serialize.rs (hunk 3 above)
Comment (fulmicoton) [reply fastfield_codecs/src/serialize.rs:218]: ah right. Actuall serialize_and_load_given_codecs is not used so I removed it.

**Review by PSeitz — APPROVED** [review]

**Pull request merged**